Signal/src/ViewControllers/DebugUI/DebugUISyncMessages.m

```diff
@@ -99,8 +99,13 @@ + (void)sendContactsSyncMessage
         [[OWSSyncContactsMessage alloc] initWithSignalAccounts:self.contactsManager.signalAccounts
                                                identityManager:self.identityManager
                                                 profileManager:self.profileManager];
-    DataSource *dataSource =
-        [DataSourceValue dataSourceWithSyncMessage:[syncContactsMessage buildPlainTextAttachmentData]];
+    __block DataSource *dataSource;
+    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
```

Comment:
NBD I'm trying to avoid these redundant `_Nonnull`.

Reply:
Yeah... ObjC nullability handling is pretty shaky when it comes to blocks, so I don't mind the redundancy. In same cases it's helpful apparently.


"redundant" nonnull provides a helpful warning

<img width="858" alt="screen shot 2018-07-05 at 1 36 51 pm" src="https://user-images.githubusercontent.com/36971200/42344175-b76edffa-8058-11e8-8ab0-76c732f89255.png">

removing "redundant" nonnull produces no warning.

<img width="869" alt="screen shot 2018-07-05 at 1 37 23 pm" src="https://user-images.githubusercontent.com/36971200/42344174-b75bfb92-8058-11e8-983a-613fb1085e97.png">

This is all within an NS_ASSUME_NONULL_BEGIN block. 🤷‍♀️